src/exodus/test/build_tx_tests.cpp

```diff
@@ -0,0 +1,159 @@
+#include "exodus/createtx.h"
+#include "exodus/errors.h"
+#include "exodus/encoding.h"
+#include "exodus/exodus.h"
+#include "exodus/wallettxs.h"
+
+#include "base58.h"
+#include "coins.h"
+#include "core_io.h"
+#include "main.h"
+#include "primitives/transaction.h"
+#include "script/script.h"
+#include "script/standard.h"
+#include "test/test_bitcoin.h"
+#include "test/fixtures.h"
+#include "utilstrencodings.h"
+#include "exodus/utilsbitcoin.h"
+
+#include "wallet/wallet.h"
+
+#include <boost/test/unit_test.hpp>
+
+#include <stdint.h>
+#include <string>
+#include <utility>
+#include <vector>
+
+BOOST_FIXTURE_TEST_SUITE(exodus_build_tx_tests, ZerocoinTestingSetup200)
+
+BOOST_AUTO_TEST_CASE(wallettxbuilder_create_normal_b)
+{
+    std::vector<unsigned char> data(nMaxDatacarrierBytes + 1);
+
+    std::string fromAddress = CBitcoinAddress(pubkey.GetID()).ToString();
+
+    uint256 txid;
+    std::string rawHex;
+    BOOST_CHECK_EQUAL(
+        0, // No error
+        exodus::WalletTxBuilder(fromAddress, "", "", 0, data, txid, rawHex, false)
+    );
+
+    CTransaction decTx;
+    BOOST_CHECK(DecodeHexTx(decTx, rawHex));
+
+    BOOST_CHECK(!decTx.IsSigmaSpend());
+
+    // calculate class b encode
+    CPubKey redeemingPubKey;
+    exodus::AddressToPubKey(fromAddress, redeemingPubKey);
+
+    std::vector<std::pair<CScript, int64_t>> vecOutputs;
+    Exodus_Encode_ClassB(fromAddress, redeemingPubKey, data, vecOutputs);
+
+    // assert scripts.
+    BOOST_CHECK_EQUAL(vecOutputs.size() + 1, decTx.vout.size());
+}
+
+BOOST_AUTO_TEST_CASE(wallettxbuilder_create_normal_c)
+{
+    std::vector<unsigned char> data(80);
+
+    std::string fromAddress = CBitcoinAddress(pubkey.GetID()).ToString();
+
+    uint256 txid;
+    std::string rawHex;
+    BOOST_CHECK_EQUAL(
+        0, // No error
+        exodus::WalletTxBuilder(fromAddress, "", "", 0, data, txid, rawHex, false)
+    );
+
+    CTransaction decTx;
+    BOOST_CHECK(DecodeHexTx(decTx, rawHex));
+
+    BOOST_CHECK(!decTx.IsSigmaSpend());
+
+    // calculate class c encode
+    std::vector<std::pair<CScript, int64_t>> vecOutputs;
+    Exodus_Encode_ClassC(data, vecOutputs);
+
+    // assert script.
+    BOOST_CHECK_EQUAL(vecOutputs.size() + 1, decTx.vout.size());
+}
+
+BOOST_AUTO_TEST_CASE(wallettxbuilder_create_sigma_without_mints)
+{
+    pwalletMain->SetBroadcastTransactions(true);
+    CreateAndProcessEmptyBlocks(200, scriptPubKey);
+
+    std::vector<unsigned char> data(nMaxDatacarrierBytes + 1, 0x01);
```

Comment:
I think we should use data that capable encoded in class C since we want to test the other cases.